cluster/nginx.go

```diff
@@ -143,15 +131,26 @@ func DeleteNginxLBDeployment(clientset *kubernetes.Clientset, deploymentName, ap
 
 		go deploymentInformer.Run(doneCh)
 
-		// This delete option is to make sure that the pods belonging to
-		// the nginx-resolver deployment are 'collected' immediately.
-		fgPropagation := metav1.DeletePropagationForeground
-		fgDeleteOption := metav1.DeleteOptions{
-			PropagationPolicy: &fgPropagation,
-		}
-		err := extV1beta1API(clientset).Deployments("default").Delete(deploymentName, &fgDeleteOption)
+		labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"type": "nginx-resolver"}}
+		deployments, err := extV1beta1API(clientset).Deployments("default").List(metav1.ListOptions{
+			LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+		})
 		if err != nil {
-			close(doneCh) // the informer listening for delete deployment event needs to be stopped
+			close(doneCh)
+		}
+		for _, deployment := range deployments.Items {
+			if deployment.Name != deploymentName {
+				// This delete option is to make sure that the pods belonging to
+				// the nginx-resolver deployment are 'collected' immediately.
+				fgPropagation := metav1.DeletePropagationForeground
+				fgDeleteOption := metav1.DeleteOptions{
+					PropagationPolicy: &fgPropagation,
+				}
+				err := extV1beta1API(clientset).Deployments("default").Delete(deployment.Name, &fgDeleteOption)
```

Comment:
I'm wondering if only the first element of the `for` will be considered since once the Delete of the first Item finishes it will close the channel on the Informer (line 127), therefore continuing with the wait code on `DeployNginxResolver` line 193. is that correct?